terraform/gitlab/runner/Dockerfile

```diff
@@ -1,5 +1,5 @@
 ARG registry
-FROM ${registry}docker.io/library/docker:20.10.18
+FROM ${registry}docker.io/library/docker:24.0.2
 
 RUN apk -U add make bash docker-credential-ecr-login
```

Comment:
I think we need `apk update` and `upgrade` here as well. And the ARG that forces the layer, as in the other PR.